### PR TITLE
Add NoReleaseAvailable callback

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests if this modifies the Windows code?~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

- Update Xamarin bindings for new onNoReleaseAvailable callback.
- Update test apps.

## Related PRs or issues

[AB#84055](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/84055)

### CODE REVIEW

File: SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.iOS.Bindings/ApiDefinition.cs [hunk 1]
```diff
@@ -78,6 +78,10 @@ interface MSACDistributeDelegate
         // @optional - (BOOL)distribute:(MSACDistribute *)distribute releaseAvailableWithDetails:(MSACReleaseDetails *)details;
         [Export("distribute:releaseAvailableWithDetails:")]
         bool OnReleaseAvailable(MSACDistribute distribute, MSACReleaseDetails details);
+
+        // - (void)noReleaseAvailable;
+        [Export("noReleaseAvailable")]
+        void NoReleaseAvailable();
```
Comment (DmitriyKirakosyan) [inline SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.iOS.Bindings/ApiDefinition.cs:84]: I think it should be `OnNoReleaseAvailable(MSACDistribute distribute)`.

File: SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.Android/Distribute.cs [hunk 2]
```diff
@@ -126,6 +128,27 @@ public bool OnReleaseAvailable(Activity activity, AndroidReleaseDetails androidR
                 }
                 return false;
             }
+
+            public void OnNoReleaseAvailable(Activity activity)
+            {
+                if (_noReleaseAvailableCallback != null)
```
Comment (DmitriyKirakosyan) [inline SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.Android/Distribute.cs:134]: `_noReleaseAvailableCallback?.Invoke`?

File: SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.Android/Distribute.cs [hunk 3]
```diff
@@ -126,6 +128,27 @@ public bool OnReleaseAvailable(Activity activity, AndroidReleaseDetails androidR
...
+                {
+                    _noReleaseAvailableCallback();
+                }
+            }
+        }
+
+        static void SetNoReleaseAvailable(NoReleaseAvailableCallback noReleaseAvailable)
```
Comment (DmitriyKirakosyan) [inline SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.Android/Distribute.cs:141]: Let's keep the class `Listener` on the bottom and move this method right under the `SetReleaseAvailableCallback` one.

File: SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.iOS.Bindings/ApiDefinition.cs (hunk 1 above)
Comment (AnastasiaKubova) [reply SDK/AppCenterDistribute/Microsoft.AppCenter.Distribute.iOS.Bindings/ApiDefinition.cs:84]: Nope, it should be the same as in apple SDK.

File: Apps/Contoso.Android.Puppet/MainActivity.cs [hunk 4]
```diff
@@ -48,12 +48,13 @@ protected override void OnCreate(Bundle savedInstanceState)
             Crashes.SendingErrorReport += SendingErrorReportHandler;
             Crashes.SentErrorReport += SentErrorReportHandler;
             Crashes.FailedToSendErrorReport += FailedToSendErrorReportHandler;
-
+
```
Comment (russelarms) [inline Apps/Contoso.Android.Puppet/MainActivity.cs:51]:
```suggestion
```

**Review by russelarms — APPROVED** [review]

**Review by DmitriyKirakosyan — APPROVED** [review]

**Review by russelarms — APPROVED** [review]

**Pull request merged**